edu.cuny.hunter.streamrefactoring.core/src/edu/cuny/hunter/streamrefactoring/core/analysis/Util.java

```diff
@@ -761,4 +785,48 @@ private static boolean wouldOrderingBeConsistent(final Collection<TypeAbstractio
 
 	private Util() {
 	}
+
+	public static boolean isStreamNode(CGNode node, IClassHierarchy classHierarchy) {
+		if (checkDeclaredClass(node))
+			return true;
+
+		IR ir = node.getIR();
+
+		if (ir == null || ir.isEmptyIR())
+			return true;
+
+		for (SSAInstruction instruction : ir.getInstructions()) {
+			if (instruction == null)
+				continue;
+
+			StreamFindingVisitor visitor = new StreamFindingVisitor(classHierarchy);
+			instruction.visit(visitor);
+
+			if (visitor.hasFoundStream())
+				return true;
+
+			// otherwise, let's check the defs and uses.
+			TypeInference inference = TypeInference.make(ir, false);
+
+			Stream<TypeAbstraction> defs = IntStream.range(0, instruction.getNumberOfDefs())
+					.mapToObj(i -> instruction.getDef(i)).flatMap(d -> getPossibleStreamTypes(d, inference).stream());
+
+			Stream<TypeAbstraction> uses = IntStream.range(0, instruction.getNumberOfUses())
+					.mapToObj(i -> instruction.getUse(i)).flatMap(u -> getPossibleStreamTypes(u, inference).stream());
+
+			if (Stream.concat(defs, uses).anyMatch(t -> implementsBaseStream(t.getTypeReference(), classHierarchy)))
+				return true;
+		}
+
+		return false;
+	}
+
+	// check declared class for CGNode
+	private static boolean checkDeclaredClass(CGNode node) {
+		IMethod method = node.getMethod();
+		if (isBaseStream(method.getDeclaringClass()))
```

Comment:
Why not implementsBaseStream?

Reply:
Because the types of the parameters for `implementsBaseStream` should be ITypebinding or TypeReference, but the parameter type for isBaseStream is `IClass`, which means I can directly use it.

Reply:
Use the `IClass` to get the `TypeReference`. I think you can use the `IClassHierarchy` to do so but don't quote me.